rust/polars-ocaml-macros/src/lib.rs

```diff
@@ -65,14 +69,19 @@ fn ocaml_interop_export_implementation(item_fn: syn::ItemFn) -> TokenStream2 {
     let native_function = quote! {
         #[no_mangle]
         pub extern "C" #signature {
-            let #runtime_name = unsafe { &mut ::ocaml_interop::OCamlRuntime::recover_handle() };
+            match ::std::panic::catch_unwind(|| {
+                let #runtime_name = unsafe { &mut ::ocaml_interop::OCamlRuntime::recover_handle() };
```

Comment:
Just curious, is this how `ocaml-interop` passes the runtime?

Reply:
`ocaml-interop` doesn't have any panic-catching code at all, but yeah, the previous version of the code should pretty much match what ocaml-interop's macros expand out to. Do you see any issues with the expansion?

Reply:
I was just curious as this uses `unsafe`. If this is what ocaml-interop does, then it should be safe

Reply:
Ah right, this is fine, since this is marked unsafe just to prevent folks from (ab)using it. See discussions [here](https://docs.rs/ocaml-interop/latest/ocaml_interop/struct.OCamlRuntime.html#method.recover_handle) and https://github.com/tizoc/ocaml-interop/issues/42 for more details